core/src/main/java/org/elasticsearch/index/query/TermQueryParser.java

```diff
@@ -63,6 +63,9 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 // skip
             } else if (token == XContentParser.Token.START_OBJECT) {
                 // also support a format of "term" : {"field_name" : { ... }}
+                if (fieldName != null) {
+                    throw new QueryParsingException(parseContext, "[term] query does not support different field names, use [terms] query");
```

Comment:
I would modify the error message here, one cannot use the terms query instead against multiple fields, you should rather combine multiple term queries in a bool query I think